include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -505,7 +505,7 @@ __radix_sort_reorder_submit(_ExecutionPolicy&& __exec, ::std::size_t __segments,
                 // find offsets for the same values within a segment and fill the resulting buffer
                 for (::std::size_t __val_idx = __seg_start + __self_lidx; __val_idx < __seg_end; __val_idx += __sg_size)
                 {
-                    auto __in_val = __input_rng[__val_idx];
+                    auto&& __in_val = __input_rng[__val_idx]; //we use C++ forwarding reference here
```

Comment:
Is this enough though to keep the constraints at minimum? In particular, don't we still need to explicitly move at line 522?

Reply:
Yes, it makes sense to keep the same "move" approach.